[Bug]: When the Metastore type is Hadoop, Terminal executes the spark sql without loading the configuration file

### What happened?

When I configure a Metastore type as a Hadoop catalog, I cannot execute sql in Terminal while ensuring the configuration file is correct

### Affects Versions

0.4.x

### What engines are you seeing the problem on?

AMS, Spark

### How to reproduce

1.Create a catalog with Metastore Type is Hadoop and ensure that the hdfs-site and core-site are connected to multiple nodes via nameservice
2.Execute sql in terminal

### Relevant log output

```shell
2023/02/09 17:04:11 new sql script submit, current thread pool state. [Active: 1, PoolSize: 1]
2023/02/09 17:04:11 fetch terminal session: node01diklwr62xump177z9vrs1dbzb0.node0-KERBEROS-*******************-iceberg_catalog_test
2023/02/09 17:04:11 session configuration: spark.sql.catalog.native_iceberg_catalog_test => org.apache.iceberg.spark.SparkCatalog
2023/02/09 17:04:11 session configuration: spark.sql.catalog.native_iceberg_catalog_test.table-formats => ICEBERG
2023/02/09 17:04:11 session configuration: spark.sql.catalog.native_iceberg_catalog_test.type => hadoop
2023/02/09 17:04:11 session configuration: spark.sql.catalog.native_iceberg_catalog_test.warehouse => hdfs://nameservice/user/arctic/hive_db/arctic.db/native_iceberg_test/
2023/02/09 17:04:11 session configuration: spark.sql.arctic.refresh-catalog-before-usage => true
2023/02/09 17:04:11  
2023/02/09 17:04:11 prepare execute statement, line:1
2023/02/09 17:04:11 show databases
2023/02/09 17:04:12 meet exception during execution.
2023/02/09 17:04:12 java.lang.IllegalArgumentException: java.net.UnknownHostException: nameservice
	at org.apache.hadoop.security.SecurityUtil.buildTokenService(SecurityUtil.java:444)
	at org.apache.hadoop.hdfs.NameNodeProxiesClient.createProxyWithClientProtocol(NameNodeProxiesClient.java:132)
	at org.apache.hadoop.hdfs.DFSClient.(DFSClient.java:353)
	at org.apache.hadoop.hdfs.DFSClient.(DFSClient.java:287)
	at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:134)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3242)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:121)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3291)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3259)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:470)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:356)
	at org.apache.iceberg.hadoop.Util.getFs(Util.java:51)
	at org.apache.iceberg.hadoop.HadoopCatalog.initialize(HadoopCatalog.java:106)
	at org.apache.iceberg.CatalogUtil.loadCatalog(CatalogUtil.java:197)
	at org.apache.iceberg.CatalogUtil.buildIcebergCatalog(CatalogUtil.java:234)
	at org.apache.iceberg.spark.SparkCatalog.buildIcebergCatalog(SparkCatalog.java:119)
	at org.apache.iceberg.spark.SparkCatalog.initialize(SparkCatalog.java:411)
	at org.apache.spark.sql.connector.catalog.Catalogs$.load(Catalogs.scala:61)
	at org.apache.spark.sql.connector.catalog.CatalogManager.$anonfun$catalog$1(CatalogManager.scala:52)
	at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:86)
	at org.apache.spark.sql.connector.catalog.CatalogManager.catalog(CatalogManager.scala:52)
	at org.apache.spark.sql.connector.catalog.LookupCatalog$CatalogAndNamespace$.unapply(LookupCatalog.scala:92)
	at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs$$anonfun$apply$1.applyOrElse(ResolveCatalogs.scala:209)
	at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs$$anonfun$apply$1.applyOrElse(ResolveCatalogs.scala:34)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsDown$2(AnalysisHelper.scala:108)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:73)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsDown$1(AnalysisHelper.scala:108)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:221)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsDown(AnalysisHelper.scala:106)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsDown$(AnalysisHelper.scala:104)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperators(AnalysisHelper.scala:73)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperators$(AnalysisHelper.scala:72)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperators(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs.apply(ResolveCatalogs.scala:34)
	at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs.apply(ResolveCatalogs.scala:29)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$2(RuleExecutor.scala:216)
	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
	at scala.collection.immutable.List.foldLeft(List.scala:89)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1(RuleExecutor.scala:213)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1$adapted(RuleExecutor.scala:205)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:205)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.org$apache$spark$sql$catalyst$analysis$Analyzer$$executeSameContext(Analyzer.scala:196)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:190)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:155)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:183)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:88)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.executeAndTrack(RuleExecutor.scala:183)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:174)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:228)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:173)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:73)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:143)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:772)
	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:143)
	at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:73)
	at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:71)
	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:63)
	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:98)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:772)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:96)
	at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:615)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:772)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:610)
	at com.netease.arctic.ams.server.terminal.local.LocalTerminalSession.executeStatement(LocalTerminalSession.java:60)
	at com.netease.arctic.ams.server.terminal.TerminalSessionContext$ExecutionTask.executeStatement(TerminalSessionContext.java:275)
	at com.netease.arctic.ams.server.terminal.TerminalSessionContext$ExecutionTask.execute(TerminalSessionContext.java:238)
	at com.netease.arctic.ams.server.terminal.TerminalSessionContext$ExecutionTask.lambda$get$0(TerminalSessionContext.java:199)
	at com.netease.arctic.table.TableMetaStore.doAsUgi(TableMetaStore.java:355)
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:335)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1873)
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:335)
	at com.netease.arctic.ams.server.terminal.TerminalSessionContext$ExecutionTask.get(TerminalSessionContext.java:191)
	at com.netease.arctic.ams.server.terminal.TerminalSessionContext$ExecutionTask.get(TerminalSessionContext.java:162)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.UnknownHostException: nameservice
	... 83 more
```


### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalManager.java
+++ b/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalManager.java
@@ -47,6 +47,8 @@ import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.stream.Collectors;
 
+import static com.netease.arctic.ams.api.properties.CatalogMetaProperties.TABLE_FORMATS;
+
 public class TerminalManager {
 
   private static final Logger LOG = LoggerFactory.getLogger(TerminalManager.class);
@@ -61,6 +63,7 @@ public class TerminalManager {
 
   private final Object sessionMapLock = new Object();
   private final Map<String, TerminalSessionContext> sessionMap = Maps.newHashMap();
+
 
   ThreadPoolExecutor executionPool = new ThreadPoolExecutor(
       1, 50, 30, TimeUnit.MINUTES,
@@ -88,9 +91,11 @@ public class TerminalManager {
   public String executeScript(String terminalId, String catalog, String script) {
     Optional<CatalogMeta> optCatalogMeta = ServiceContainer.getCatalogMetadataService().getCatalog(catalog);
     if (!optCatalogMeta.isPresent()) {
-      throw new IllegalArgumentException("catalog " + catalog + " is not validea");
+      throw new IllegalArgumentException("catalog " + catalog + " is not validate");
     }
     CatalogMeta catalogMeta = optCatalogMeta.get();
+    boolean isNativeIceberg = catalogMeta.getCatalogProperties().containsKey(TABLE_FORMATS) &&
+        catalogMeta.catalogProperties.get(TABLE_FORMATS).equalsIgnoreCase("iceberg");
     TableMetaStore metaStore = getCatalogTableMetaStore(catalogMeta);
     String sessionId = getSessionId(terminalId, metaStore, catalog);
     String catalogType = CatalogUtil.isIcebergCatalog(catalog) ? "iceberg" : "arctic";
@@ -103,6 +108,7 @@ public class TerminalManager {
       String value = catalogMeta.getCatalogProperties().get(key);
       configuration.set(SessionConfigOptions.catalogProperty(catalog, key), value);
     }
+    configuration.set(SessionConfigOptions.IS_NATIVE_ICEBERG, isNativeIceberg);
     configuration.set(SessionConfigOptions.catalogProperty(catalog, "type"),
         catalogMeta.getCatalogType());
 

--- a/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalSessionFactory.java
+++ b/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalSessionFactory.java
@@ -65,6 +65,11 @@ public interface TerminalSessionFactory {
         .stringType()
         .noDefaultValue();
 
+    public static ConfigOption<Boolean> IS_NATIVE_ICEBERG = ConfigOptions
+        .key("is-native-iceberg")
+        .booleanType()
+        .noDefaultValue();
+
     public static ConfigOption<String> catalogConnector(String catalog) {
       return ConfigOptions.key("session.catalog." + catalog + ".connector")
           .stringType()

--- a/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/local/LocalSessionFactory.java
+++ b/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/local/LocalSessionFactory.java
@@ -48,7 +48,6 @@ public class LocalSessionFactory implements TerminalSessionFactory {
   );
 
   SparkSession context = null;
-  SparkSession nativeIcebergSession = null;
 
   Configuration conf = null;
 
@@ -61,7 +60,7 @@ public class LocalSessionFactory implements TerminalSessionFactory {
   public TerminalSession create(TableMetaStore metaStore, Configuration configuration) {
     Boolean isNativeIceberg = configuration.get(SessionConfigOptions.IS_NATIVE_ICEBERG);
     List<String> catalogs = configuration.get(SessionConfigOptions.CATALOGS);
-    SparkSession context = lazyInitContext(conf, metaStore, isNativeIceberg, catalogs.get(0));
+    SparkSession context = lazyInitContext(conf);
     SparkSession session = context.cloneSession();
     List<String> initializeLogs = Lists.newArrayList();
     initializeLogs.add("initialize session, session factory: " + LocalSessionFactory.class.getName());
@@ -69,6 +68,12 @@ public class LocalSessionFactory implements TerminalSessionFactory {
     Map<String, String> sparkConf = SparkContextUtil.getSparkConf(configuration);
     Map<String, String> finallyConf = Maps.newLinkedHashMap();
     sparkConf.put(REFRESH_CATALOG_BEFORE_USAGE, "true");
+    if (isNativeIceberg) {
+      org.apache.hadoop.conf.Configuration metaConf = metaStore.getConfiguration();
+      for (Map.Entry<String, String> next : metaConf) {
+        session.conf().set("spark.sql.catalog." + catalogs.get(0) + ".hadoop." + next.getKey(), next.getValue());
+      }
+    }
     for (String key : sparkConf.keySet()) {
       if (STATIC_SPARK_CONF.contains(key)) {
         continue;
@@ -85,53 +90,30 @@ public class LocalSessionFactory implements TerminalSessionFactory {
     logs.add(key + "  " + value);
   }
 
-  protected synchronized SparkSession lazyInitContext(Configuration conf, TableMetaStore metaStore,
-                                                      Boolean isNativeIceberg, String catalog) {
-    if (!isNativeIceberg) {
-      if (context == null) {
-        SparkConf sparkConf = buildSparkBasicConf(conf);
-        context = SparkSession
-            .builder()
-            .config(sparkConf)
-            .getOrCreate();
-        context.sparkContext().setLogLevel("WARN");
-      }
-
-      return context;
-    } else {
-      if (nativeIcebergSession == null) {
-        SparkConf sparkConf = buildSparkBasicConf(conf);
-        org.apache.hadoop.conf.Configuration configuration = metaStore.getConfiguration();
-        for (Map.Entry<String, String> next : configuration) {
-          sparkConf.set("spark.sql.catalog." + catalog + ".hadoop." + next.getKey(), next.getValue());
+  protected synchronized SparkSession lazyInitContext(Configuration conf) {
+    if (context == null) {
+      SparkConf sparkconf = new SparkConf()
+          .setAppName("spark-local-context")
+          .setMaster("local");
+      sparkconf.set(SQLConf.PARTITION_OVERWRITE_MODE().key(), "dynamic");
+      sparkconf.set("spark.executor.heartbeatInterval", "100s");
+      sparkconf.set("spark.network.timeout", "200s");
+      sparkconf.set("spark.sql.extensions", ArcticSparkExtensions.class.getName() +
+          "," + IcebergSparkSessionExtensions.class.getName());
+      for (String key : conf.keySet()) {
+        if (!key.startsWith(ArcticMetaStoreConf.SPARK_CONF)) {
+          continue;
         }
-        nativeIcebergSession = SparkSession
-            .builder()
-            .config(sparkConf)
-            .getOrCreate();
-        nativeIcebergSession.sparkContext().setLogLevel("WARN");
+        String value = conf.getValue(ConfigOptions.key(key).stringType().noDefaultValue());
+        sparkconf.set(key, value);
       }
-      return nativeIcebergSession;
+      context = SparkSession
+          .builder()
+          .config(sparkconf)
+          .getOrCreate();
+      context.sparkContext().setLogLevel("WARN");
     }
 
-  }
-
-  public SparkConf buildSparkBasicConf(Configuration conf) {
-    SparkConf sparkconf = new SparkConf()
-        .setAppName("spark-local-context")
-        .setMaster("local");
-    sparkconf.set(SQLConf.PARTITION_OVERWRITE_MODE().key(), "dynamic");
-    sparkconf.set("spark.executor.heartbeatInterval", "100s");
-    sparkconf.set("spark.network.timeout", "200s");
-    sparkconf.set("spark.sql.extensions", ArcticSparkExtensions.class.getName() +
-        "," + IcebergSparkSessionExtensions.class.getName());
-    for (String key : conf.keySet()) {
-      if (!key.startsWith(ArcticMetaStoreConf.SPARK_CONF)) {
-        continue;
-      }
-      String value = conf.getValue(ConfigOptions.key(key).stringType().noDefaultValue());
-      sparkconf.set(key, value);
-    }
-    return sparkconf;
+    return context;
   }
 }

--- a/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/local/LocalSessionFactory.java
+++ b/ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/local/LocalSessionFactory.java
@@ -48,6 +48,7 @@ public class LocalSessionFactory implements TerminalSessionFactory {
   );
 
   SparkSession context = null;
+  SparkSession nativeIcebergSession = null;
 
   Configuration conf = null;
 
@@ -58,9 +59,10 @@ public class LocalSessionFactory implements TerminalSessionFactory {
 
   @Override
   public TerminalSession create(TableMetaStore metaStore, Configuration configuration) {
-    SparkSession context = lazyInitContext(conf);
-    SparkSession session = context.cloneSession();
+    Boolean isNativeIceberg = configuration.get(SessionConfigOptions.IS_NATIVE_ICEBERG);
     List<String> catalogs = configuration.get(SessionConfigOptions.CATALOGS);
+    SparkSession context = lazyInitContext(conf, metaStore, isNativeIceberg, catalogs.get(0));
+    SparkSession session = context.cloneSession();
     List<String> initializeLogs = Lists.newArrayList();
     initializeLogs.add("initialize session, session factory: " + LocalSessionFactory.class.getName());
 
@@ -83,30 +85,53 @@ public class LocalSessionFactory implements TerminalSessionFactory {
     logs.add(key + "  " + value);
   }
 
-  protected synchronized SparkSession lazyInitContext(Configuration conf) {
-    if (context == null) {
-      SparkConf sparkconf = new SparkConf()
-          .setAppName("spark-local-context")
-          .setMaster("local");
-      sparkconf.set(SQLConf.PARTITION_OVERWRITE_MODE().key(), "dynamic");
-      sparkconf.set("spark.executor.heartbeatInterval", "100s");
-      sparkconf.set("spark.network.timeout", "200s");
-      sparkconf.set("spark.sql.extensions", ArcticSparkExtensions.class.getName() +
-          "," + IcebergSparkSessionExtensions.class.getName());
-      for (String key : conf.keySet()) {
-        if (!key.startsWith(ArcticMetaStoreConf.SPARK_CONF)) {
-          continue;
-        }
-        String value = conf.getValue(ConfigOptions.key(key).stringType().noDefaultValue());
-        sparkconf.set(key, value);
+  protected synchronized SparkSession lazyInitContext(Configuration conf, TableMetaStore metaStore,
+                                                      Boolean isNativeIceberg, String catalog) {
+    if (!isNativeIceberg) {
+      if (context == null) {
+        SparkConf sparkConf = buildSparkBasicConf(conf);
+        context = SparkSession
+            .builder()
+            .config(sparkConf)
+            .getOrCreate();
+        context.sparkContext().setLogLevel("WARN");
       }
-      context = SparkSession
-          .builder()
-          .config(sparkconf)
-          .getOrCreate();
-      context.sparkContext().setLogLevel("WARN");
+
+      return context;
+    } else {
+      if (nativeIcebergSession == null) {
+        SparkConf sparkConf = buildSparkBasicConf(conf);
+        org.apache.hadoop.conf.Configuration configuration = metaStore.getConfiguration();
+        for (Map.Entry<String, String> next : configuration) {
+          sparkConf.set("spark.sql.catalog." + catalog + ".hadoop." + next.getKey(), next.getValue());
+        }
+        nativeIcebergSession = SparkSession
+            .builder()
+            .config(sparkConf)
+            .getOrCreate();
+        nativeIcebergSession.sparkContext().setLogLevel("WARN");
+      }
+      return nativeIcebergSession;
     }
 
-    return context;
+  }
+
+  public SparkConf buildSparkBasicConf(Configuration conf) {
+    SparkConf sparkconf = new SparkConf()
+        .setAppName("spark-local-context")
+        .setMaster("local");
+    sparkconf.set(SQLConf.PARTITION_OVERWRITE_MODE().key(), "dynamic");
+    sparkconf.set("spark.executor.heartbeatInterval", "100s");
+    sparkconf.set("spark.network.timeout", "200s");
+    sparkconf.set("spark.sql.extensions", ArcticSparkExtensions.class.getName() +
+        "," + IcebergSparkSessionExtensions.class.getName());
+    for (String key : conf.keySet()) {
+      if (!key.startsWith(ArcticMetaStoreConf.SPARK_CONF)) {
+        continue;
+      }
+      String value = conf.getValue(ConfigOptions.key(key).stringType().noDefaultValue());
+      sparkconf.set(key, value);
+    }
+    return sparkconf;
   }
 }

--- a/spark/v3.1/spark/src/main/java/com/netease/arctic/spark/reader/UnkeyedSparkBatchScan.java
+++ b/spark/v3.1/spark/src/main/java/com/netease/arctic/spark/reader/UnkeyedSparkBatchScan.java
@@ -123,7 +123,8 @@ public class UnkeyedSparkBatchScan implements Scan, Batch, SupportsReportStatist
     if (table.currentSnapshot() == null) {
       return new Stats(0L, 0L);
     }
-    if (!table.spec().isUnpartitioned() && filterExpressions.isEmpty()) {
+    if (!table.spec().isUnpartitioned() &&
+        (filterExpressions == null || filterExpressions.isEmpty())) {
       LOG.debug("using table metadata to estimate table statistics");
       long totalRecords =
           PropertyUtil.propertyAsLong(

